<a href="https://colab.research.google.com/github/v-3sg/sales_predictions/blob/main/sales_predictions.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
filename = "/content/drive/MyDrive/Colab Notebooks/Coding Dojo/DS Week 1/sales_predictions.csv"


In [2]:
df = pd.read_csv(filename)

1 - Explore the data - what do you need to do to clean this data? Clean it!



Data Dictionary for this dataset:

Variable Name	Description

Item_Identifier	Unique product ID

Item_Weight	Weight of product

Item_Fat_Content	Whether the product is low fat or regular

Item_Visibility	The percentage of total display area of all products in a store allocated to the particular product

Item_Type	The category to which the product belongs

Item_MRP	Maximum Retail Price (list price) of the product

Outlet_Identifier	Unique store ID

Outlet_Establishment_Year	The year in which store was established

Outlet_Size	The size of the store in terms of ground area covered

Outlet_Location_Type	The type of area in which the store is located

Outlet_Type	Whether the outlet is a grocery store or some sort of supermarket

Item_Outlet_Sales	Sales of the product in the particular store. This is the target variable to be predicted.


**--> Based on the Data Dictionary, potential irrelevant columns include:**

**Outlet_Establishment_Year**

**Outlet_Size**

**Outlet_Location**

**Outlet_Type**

**... However, there is the possibility that these columns represent variables that may affect whether a certain food item is purchased, e.g., an outlet of a certain size/type/age may nudge a customer to buy an item. My initial thought is not to drop any columns.**


In [3]:
#check for duplicates
df.duplicated().any()

False

**--> No duplicates present**

2- Are there missing values in this dataset? Some stores might not report all the data due to technical glitches or other issues. If so, deal with these appropriately.

In [4]:
#explore the data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [5]:
#check for NaNs/nulls
df.isnull().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

**--> Confirmed count of null values for Item_Weight (1463) and Outlet_Size (2410)**

3 - What are some summary statistics, aggregate information, or other useful trends you can see from the data using Pandas?

In [6]:
df.head()

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
0           FDA15         9.30  ...  Supermarket Type1          3735.1380
1           DRC01         5.92  ...  Supermarket Type2           443.4228
2           FDN15        17.50  ...  Supermarket Type1          2097.2700
3           FDX07        19.20  ...      Grocery Store           732.3800
4           NCD19         8.93  ...  Supermarket Type1           994.7052

[5 rows x 12 columns]

In [7]:
df.tail()

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
8518           FDF22        6.865  ...  Supermarket Type1          2778.3834
8519           FDS36        8.380  ...  Supermarket Type1           549.2850
8520           NCJ29       10.600  ...  Supermarket Type1          1193.1136
8521           FDN46        7.210  ...  Supermarket Type2          1845.5976
8522           DRG01       14.800  ...  Supermarket Type1           765.6700

[5 rows x 12 columns]

In [8]:
#explore descriptive statistics of data
df.describe(include = 'all')

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
count             8523  7060.000000  ...               8523        8523.000000
unique            1559          NaN  ...                  4                NaN
top              FDW13          NaN  ...  Supermarket Type1                NaN
freq                10          NaN  ...               5577                NaN
mean               NaN    12.857645  ...                NaN        2181.288914
std                NaN     4.643456  ...                NaN        1706.499616
min                NaN     4.555000  ...                NaN          33.290000
25%                NaN     8.773750  ...                NaN         834.247400
50%                NaN    12.600000  ...                NaN        1794.331000
75%                NaN    16.850000  ...                NaN        3101.296400
max                NaN    21.350000  ...                NaN       13086.964800

[11 rows x 12 columns]

**--> Summary of statistics:**

**There seems to be a weight outlier (min of Item_Weight = 4.555). Note Item_Weight data may be skewed due to presence of NaN values for this column.**

**Low Fat items seemed to be the most bought (Item_Fat_Content frequency = 5089)**

**Average price of purchased items was about 140.99 (see Item_MRP). Possible outliers for min value and max value.**

**For Outlet_Size, frequency of Medium was highest (2793); however, this may be skewed due to presence of NaN values for this column.**

**For Outlet_Location_Type, frequency of Tier 3 was highest (3350).**

**For Outlet_Type, frequency of Supermarket Type 1 was highest (5577).**

**For Item_Outlet_Sales, there seems to be outliers with min at 33.29 and max at 13086.96**




In [9]:
#look at sample of null values
null_df = df[df.isnull().any(axis=1)]
null_df

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
3              FDX07        19.20  ...      Grocery Store           732.3800
7              FDP10          NaN  ...  Supermarket Type3          4022.7636
8              FDH17        16.20  ...  Supermarket Type1          1076.5986
9              FDU28        19.20  ...  Supermarket Type1          4710.5350
18             DRI11          NaN  ...  Supermarket Type3          2303.6680
...              ...          ...  ...                ...                ...
8504           NCN18          NaN  ...  Supermarket Type3          4138.6128
8508           FDW31        11.35  ...  Supermarket Type1          2587.9646
8509           FDG45         8.10  ...      Grocery Store           424.7804
8514           FDA01        15.00  ...  Supermarket Type1           468.7232
8519           FDS36         8.38  ...  Supermarket Type1           549.2850

[3873 rows x 12 columns]

**--> Cannot determine any pattern with the NaN values**